src/awkward/_nplikes/dispatch.py

```diff
@@ -44,6 +45,11 @@ def nplike_of_obj(
     except KeyError:
         # Try and find the nplike for this type
         # caching the result by type
+        # For virtual arrays, this is checked in the except block
+        # to avoid the isinstance slowdown inside the try block
+        # because it's used often for non-virtual arrays
```

Comment:
```suggestion
        # because it's used often for non-virtual arrays
        # TODO: replace this whole function with a more generic lookup registration system
```